src/bp/nlu/stan/model-repo.ts

```diff
@@ -1,31 +1,90 @@
 import crypto from 'crypto'
 import fse, { WriteStream } from 'fs-extra'
 import _ from 'lodash'
-import mkdirp from 'mkdirp'
 import * as NLUEngine from 'nlu/engine'
 import path from 'path'
+import Logger from 'simple-logger'
 import { Stream } from 'stream'
 import tar from 'tar'
 import tmp from 'tmp'
+import {
+  Database,
+  DBStorageDriver,
+  DiskStorageDriver,
+  GhostService,
+  ScopedGhostService,
+  MemoryObjectCache
+} from './simple-ghost'
+interface FSDriver {
+  driver: 'fs'
+}
+
+interface DBDriver {
+  driver: 'db'
+  dbURL: string
+}
+
+export type ModelRepoOptions = (FSDriver | DBDriver) & {
+  modelDir: string
+}
+
+interface ModelOwnershipOptions {
+  appId?: string
+  appSecret?: string
+}
 
-export default class ModelRepository {
-  constructor(private modelDir: string) {}
+const MODELS_DIR = './models'
+const debug = DEBUG('nlu').sub('model-repo')
+const defaultOtpions: ModelRepoOptions = {
+  modelDir: path.join(process.cwd(), 'botpress-nlu'),
+  driver: 'fs'
+}
 
-  public async init() {
-    mkdirp.sync(this.modelDir)
+export class ModelRepository {
+  private _ghost: GhostService
+  private _db: Database
+  private options: ModelRepoOptions
+
+  constructor(options: Partial<ModelRepoOptions>) {
+    this.options = { ...defaultOtpions, ...options } as ModelRepoOptions
+    const logger = new Logger('ModelRepo')
```

Comment:
I would probably pass the logger at ctor as there is an alreay instanciated one in `api.ts`.

Reply:
I did that first. I wasn't sure of the loggee name. But yes will do